### PR TITLE
Synthetic Data / Backend Added

### PR DESCRIPTION
**Pull request by Karthi-DStech**
Hi all, 

I have added backend scripts which is required for creating synthetic data. 

In the following PR, I'll implement the streamlit part. 

Kindly review and do the needful. 

Thanks and regards, 
Karthik

### CODE REVIEW

**Review by dcl10 — CHANGES_REQUESTED** [review]
I would also add a new options class like those in `biofefi.options` since we're migrating to using those, going forward.

File: biofefi/machine_learning/data.py [hunk 1]
```diff
@@ -26,6 +27,7 @@ def __init__(self, opt: argparse.Namespace, logger: object = None) -> None:
         self._normalization = opt.normalization
         self._numerical_cols = "all"
         self._n_bootstraps = opt.n_bootstraps
+        self._create_synthetic_data = None
```
[the remark this answers is not in the document]
Comment (Karthi-DStech) [reply biofefi/machine_learning/data.py:30]: Sure. I'll make the changes.

File: biofefi/machine_learning/data.py [hunk 2]
```diff
@@ -36,10 +38,28 @@ def _load_data(self) -> Tuple[pd.DataFrame, pd.DataFrame]:
...
+            self._logger.info(f"Loading data from {self._path}")
+            try:
+                df = pd.read_csv(self._path)
+                X = df.iloc[:, :-1]
+                y = df.iloc[:, -1]
+
+                return X, y
+
+            except Exception as e:
+                raise ValueError(f"Error loading the data through CSV file: {e}")
```
[the remark this answers is not in the document]
Comment (Karthi-DStech) [reply biofefi/machine_learning/data.py:62]: make sense. I'll implement it.

**Comment by Karthi-DStech** [comment]
> I would also add a new options class like those in `biofefi.options` since we're migrating to using those, going forward.

I have no problem implementing the options you mentioned. However, I was concerned about changing the options dynamically. 

As we can see, all the options mentioned in `biofefi.options` are static. What would be the case if we want to use the options as per custom requirements (eg: if the user needs to specify the number of features and training samples as per their requirements)? 

can we have a short discussion about this so I can implement it? 

Thanks and regards,
Karthik

**Comment by dcl10** [comment]
Hi @Karthi-DStech, so sorry for taking so long to get back to you. There's been quite a few changes to the app in the time you've been off. Let's leave this PR open until you get back and then we'll discuss how to move this forward.

**Comment by dcl10** [comment]
Hi @Karthi-DStech, I'm really sorry about this but I'm going to close this PR without merging. We recently did a large refactor of the way we handle options throughout the app. Also I have some other concerns around the design of making synthetic data. We can discuss it when you get back.

**Pull request closed without merging**